pilot/pkg/serviceregistry/kube/controller/controller_test.go

```diff
@@ -890,6 +896,436 @@ func TestController_Service(t *testing.T) {
 	}
 }
 
```

Comment:
It would be good to have tests covering namespace deletion

Reply:
good call